src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_IReDimOperation.vb

```diff
@@ -0,0 +1,1722 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics
+    Partial Public Class IOperationTests
+        Inherits SemanticModelTestBase
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem(30175, "https://github.com/dotnet/roslyn/issues/30175")>
+        Public Sub ReDimStatement_SingleClause_SimpleArray()
+            Dim source = <![CDATA[
+Imports System
+
+Friend Class [Class]
+    Public Sub M()
+        Dim intArray(1) As Integer
+        ReDim intArray(2)'BIND:"ReDim intArray(2)"
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IReDimOperation (OperationKind.ReDim, Type: null) (Syntax: 'ReDim intArray(2)')
+  Clauses(1):
+      IReDimClauseOperation (OperationKind.ReDimClause, Type: null) (Syntax: 'intArray(2)')
+        Operand: 
+          ILocalReferenceOperation: intArray (OperationKind.LocalReference, Type: System.Int32()) (Syntax: 'intArray')
+        Indices(1):
+            IBinaryOperation (BinaryOperatorKind.Add, Checked) (OperationKind.BinaryOperator, Type: System.Int32, Constant: 3, IsImplicit) (Syntax: '2')
+              Left: 
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 2) (Syntax: '2')
+              Right: 
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1, IsImplicit) (Syntax: '2')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of ReDimStatementSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem(30175, "https://github.com/dotnet/roslyn/issues/30175")>
+        Public Sub ReDimClause_SimpleArray()
+            Dim source = <![CDATA[
+Imports System
+
+Friend Class [Class]
+    Public Sub M()
+        Dim intArray(1) As Integer
+        ReDim intArray(2)'BIND:"intArray(2)"
```

Comment:
>ReDim intArray(2)'BIND:"intArray(2)" [](start = 8, length = 36)

It would also be good to test that we can extract IOperation nodes for the target and for the sizes directly.